src/main/java/org/springframework/data/solr/core/SolrTemplate.java

```diff
@@ -89,8 +90,9 @@
  * @author Joachim Uhrlass
  * @author Francisco Spaeth
  * @author Shiradwade Sateesh Krishna
+ * @author Venil Noronha
  */
-public class SolrTemplate implements SolrOperations, InitializingBean, ApplicationContextAware {
+public class SolrTemplate implements SolrOperations, MulticoreSolrOperations, InitializingBean, ApplicationContextAware {

```

Comment:
But does that make sense? We use a single `SolrTemplate`to query multiple cores defined by `SolrDocument.solrCoreName`. On the other hand we need to solve problems to write changes to correct core. This screams to CQRS.
My main fear is that we get an instable Multicore Support by "duplicating" the `SolrOperations`.